engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -983,7 +1005,7 @@ namespace dmGameSystem
         dmRender::RenderObject& ro = world->m_RenderObjects[world->m_RenderObjects.Size()-1];
 
         ro.Init();
-        ro.m_Material          = material;
+        ro.m_Material          = GetComponentMaterial(component, component->m_Resource, material_index);;
```

Comment:
We need to use the original material here to create the batches with, since the overridden material shouldn't affect what objects that should be rendered. I.e tags used for matching with the draw predicate are taken from the assigned material on the component and not from the overridden material.